proposals/0052-consensus-and-transaction-inclusion-verification.md

```diff
@@ -0,0 +1,169 @@
+
+This SIMD describes the overall design and changes required that allows users to verify that a supermajority of the validators has voted on the slot that their transaction was included in the block without fully trusting the RPC provider.
+
+This includes two main changes: 
+1) Adding a new RPC method which provides a proof that a transaction has been included in a slot and 
+2) Modifying the blockhash to be computed as a Merkle Tree and include transaction statuses
+
+This SIMD is the first step in implementing a consensus verifying client as first described in [SIMD #10](https://github.com/solana-foundation/solana-improvement-documents/pull/10) and a majority of the changes mentioned in the accepted [Simple Payment and State Verification
+proposal](https://docs.solana.com/proposals/simple-payment-and-state-verification).
+
+## Motivation
+
+Currently, for a user to validate whether their transaction is valid and included in a block it needs to trust the confirmation from the RPC. This has been a glaring attack vector for malicious actors that could lie to users if it's in their own interest. 
+
+To combat this, mature entities like exchanges run full nodes that process the entire ledger and can verify entire blocks. The downside of this is that it's very costly to run a full node which makes it inaccessible to everyday users,exposing users to potential attacks from malicious nodes. This is where diet clients come in, users run the client to verify the confirmation of their transaction without trusting the RPC.
+
+However, this is only the consensus verifying stage of the client, and with only these changes, the RPC provider can still trick users, hence we also discuss future work that will be implemented in future SIMDs to provide a fully trustless setup. 
+
+## Alternatives Considered
+
+None
+
+## New Terminology
+
+TransactionProof: A structure containing necessary information to verify if a transaction was included in the bank hash of a slot.
+
+```rs
+// new RPC struct to verify tx inclusion
+pub struct TransactionProof {
+  pub proof: Vec<Hash>,
+  pub parent_hash: Hash,
+  pub accounts_delta_hash: Hash,
+  pub signature_count_buf: [u8; 8],
+}
+```
+
+The proof variable will provide a Merkle hashpath from the transaction to the blockhash; which is then used with the other variables to compute the bankhash.
+
+## Detailed Design
+
+The protocol interaction will be as follows:
+- A user sends a transaction and it lands in slot N
+- The user requests proof that the transaction is included in slot N using the new RPC method 
+- The user verifies that the proof includes the transaction signature and a success status 
+- The user constructs the merkle tree to derive the root hash
+- The user computes the expected bankhash using the root hash, `parent_hash`, `accounts_delta_hash` and `signature_count`
+- The user retrieves the epoch’s current validator set and stake amounts from a trusted source (making this step trustless is future work) 
```

Comment:
Agreed, we have the same issue in lite-rpc, it's impossible to monitor without a full runtime. Eth2 has an interesting solution: https://github.com/ethereum/annotated-spec/blob/master/altair/sync-protocol.md

Reply:
> Agreed, we have the same issue in lite-rpc, it's currently impossible to construct without a full runtime. Eth2 has an interesting solution: https://github.com/ethereum/annotated-spec/blob/master/altair/sync-protocol.md

Yea this is what we were exploring as well, but this would need a small committee of trusted validators.

Reply:
> Agreed, we have the same issue in lite-rpc, it's currently impossible to construct without a full runtime. Eth2 has an interesting solution: https://github.com/ethereum/annotated-spec/blob/master/altair/sync-protocol.md

Yea this is what we were exploring as well, but this would need a small committee of trusted validators.